examples/cifar10/convert_cifar_data.cpp

```diff
@@ -37,9 +47,15 @@ void read_image(std::ifstream* file, int* label, char* buffer) {
 
 void convert_dataset(const string& input_folder, const string& output_folder,
     const string& db_type) {
+#ifdef USE_BOOST
   scoped_ptr<db::DB> train_db(db::GetDB(db_type));
   train_db->Open(output_folder + "/cifar10_train_" + db_type, db::NEW);
   scoped_ptr<db::Transaction> txn(train_db->NewTransaction());
+#else
+  unique_ptr<db::DB> train_db(db::GetDB(db_type));
+  train_db->Open(output_folder + "/cifar10_train_" + db_type, db::NEW);
+  unique_ptr<db::Transaction> txn(train_db->NewTransaction());
+#endif  // USE_BOOST
   // Data buffer
```

Comment:
In common.hpp, std can be defined as the aliass of boost names and then the std name can be used everywhere.
```
#ifdef USE_BOOST
#define unique_ptr scoped_ptr
#endif
```